graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -488,19 +447,19 @@ def _set_up_schema_elements_of_kind(self, class_name_to_definition, kind, class_
                 abstract = True
 
             property_name_to_descriptor = {}
-            all_property_lists = (
+            all_orientdb_property_lists = (
                 class_name_to_definition[inherited_class_name]['properties']
                 for inherited_class_name in self._inheritance_sets[class_name]
             )
             links = {EDGE_DESTINATION_PROPERTY_NAME: [], EDGE_SOURCE_PROPERTY_NAME: []}
-            for property_definition in chain.from_iterable(all_property_lists):
-                property_name = property_definition['name']
+            for orientdb_property_definition in chain.from_iterable(all_orientdb_property_lists):
+                property_name = orientdb_property_definition['name']
 
                 # The only properties we allow to be redefined are the in/out properties
                 # of edge classes. All other properties may only be defined once
                 # in the entire inheritance hierarchy of any schema class, of any kind.
                 duplication_allowed = all((
-                    property_name in allowed_duplicated_edge_property_names,
+                    property_name in set(links.keys()),
```

Comment:
What is the type of `links` here? If it's a dict, why not just `property_name in links`?